tidy3d/web/webapi.py

```diff
@@ -702,14 +702,24 @@ def estimate_cost(task_id: str) -> float:
     task = SimulationTask.get(task_id)
     if not task:
         raise ValueError("Task not found.")
-    resp = task.estimate_cost()
-    flex_unit = resp.get("flex_unit")
-    if not flex_unit:
-        log.warning(
-            "Could not get estimated cost! It will be reported in preprocessing upon "
-            "simulation run."
-        )
-    return flex_unit
+
+    task.estimate_cost()
+    task_info = get_info(task_id)
+    status = task_info.metadataStatus
+
+    # Wait for a termination status
+    while status not in ["processed", "error"]:
```

Comment:
worried about this getting stuck in an infinite loop, eg if `status` is `None` for some reason. Do you think this is an issue and should we safeguard against it?

Reply:
yea I suppose it is, except that design is kind of fragile too as we've seen. I guess maybe more concretely, if the metadata processing status keys changed in the future (eg to include "errored"), we'd just want this to never enter the loop, rather than get caught infinitely in it. So maybe I'd suggest something like
```
processing_statuses = [blah, blah, blah]

while status in processing_statuses:
    ...

if status == "processed":
    ....
elif status == "error"
   ...

return None
```
kind of thing maybe?

Reply:
@lei-flex what are the currently possible values for `metadataStatus`? And do you have any suggestion for good handling?